Marlin/src/feature/fwretract.cpp

```diff
@@ -110,8 +112,32 @@ void FWRetract::retract(const bool retracting
   #else
     constexpr bool swapping = false;
   #endif
+  */
 
-  /* // debugging
+  #if EXTRUDERS > 1
+
+    /* debugging
+      SERIAL_ECHOLNPAIR("retracting ", retracting);
+      SERIAL_ECHOLNPAIR("swapping ", swapping);
+      SERIAL_ECHOLNPAIR("active extruder ", active_extruder);
+      SERIAL_ECHOLNPAIR("retracted_swap ", retracted_swap[active_extruder]);
+      SERIAL_ECHOLNPAIR("Compare ", int(swapping && retracted_swap[active_extruder] == retracting));
+    //*/
+
+    // G11 priority to recover the long retract if activated
+    if (!retracting) swapping = retracted_swap[active_extruder];              //SERIAL_ECHOLN("A DONE");
+    // Allow G10 &  G10 S1 retract/recover only once
+    if (swapping && retracted_swap[active_extruder] == retracting) return;    //SERIAL_ECHOLN("B DONE");
+    if (!swapping && retracted[active_extruder] == retracting) return;        //SERIAL_ECHOLN("C DONE");
+
+  #else
+
+    if (retracted[active_extruder] == retracting) return;
+    const bool swapping = false;
+
+  #endif
```

Comment:
Are you absolutely certain these changes for the `FWRETRACT` feature are necessary? There have been a lot of people working on this feature with much trial-and-error, and it would be a shame to break it after we finally settled on something that seems to be working.

Reply:
Okay I do not wan to blame anyone. I doe see, there was a huge among of work on it and i do respect this. I will revert it and hold the change in my fork, because id did not work correct with my G-Code before post process the G-Code manual. Also the swap retraction had always the default retraction in addition. And with that change it works always fine with several single-nozzle multi-feeder, and with an Chimera Clone.

Is it okay if I just revert this change and open an issue to be discussed so i will create a new pull request if that is the result of the issue discussion?  

than i will follow the same procedure for all stuff that is not MPE Relevant.

I do think i shall had to follow this procedure from the beginning.

short feedback an i will do it in this way.

Reply:
I did revert fwretract.cpp

Reply:
A new PR around the `FWRETRACT` issue would be most welcome, so that we can get more testers on it.